org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest2.java

```diff
@@ -0,0 +1,652 @@
+			}
+		}
+	}
+
+	private String separateCommas(String value) {
+		if (!value.contains(","))
+			return value;
+		String result = "";
+		String rest = value;
+		String seperator = "";
+		while (!rest.isEmpty()) {
+			int commaIndex = rest.indexOf(',');
+			if (commaIndex == -1) {
+				result += seperator + rest;
+				rest = "";
+			} else {
+				int quote0Index = rest.indexOf('"');
+				if (quote0Index == -1 || commaIndex < quote0Index) {
+					result += seperator + rest.substring(0, commaIndex);
+					rest = rest.substring(commaIndex + 1);
+				} else {
+					int quote1Index = rest.indexOf('"', quote0Index + 1);
+					if (quote1Index == -1) {
+						result += seperator + rest.substring(0, commaIndex);
+						rest = rest.substring(commaIndex + 1);
+					} else {
+						commaIndex = rest.indexOf(',', quote1Index);
+						if (commaIndex == -1) {
+							result += seperator + rest;
+							rest = "";
+						} else {
+							result += seperator + rest.substring(0, commaIndex);
+							rest = rest.substring(commaIndex + 1);
+						}
+					}
+				}
+			}
+			seperator = "," + newline + " ";
+		}
+		return result;
+	}
+
+	@Override
+	public MergeableManifest2 clone() throws CloneNotSupportedException {
+		return new MergeableManifest2(this);
+	}
+
+	/**
+	 * Return a string that ensures that no line is longer then 512 characters
+	 * and lines are broken according to manifest specification.
+	 * 
+	 * @param input The buffer containing the content that should be made safe
+	 * @param newline The string to use to create newlines (usually "\n" or
+	 * "\r\n")
+	 * @return The string with no longer lines then 512, ready to be read again
+	 * by {@link MergeableManifest2}.
+	 */
+	public static String make512Safe(StringBuffer input, String newline) {
+		StringBuilder result = new StringBuilder();
+		String content = input.toString();
+		String rest = content;
+		while (!rest.isEmpty()) {
+			if (rest.contains("\n")) {
+				String line = rest.substring(0, rest.indexOf("\n"));
+				rest = rest.substring(rest.indexOf("\n") + 1);
+				if (line.length() > 1 && line.charAt(line.length() - 1) == '\r')
+					line = line.substring(0, line.length() - 1);
+				append512Safe(line, result, newline);
+			} else {
+				append512Safe(rest, result, newline);
+				break;
+			}
+		}
+		return result.toString();
+	}
+
+	private static String append512Safe(String toAppend, StringBuilder result, String newline) {
+		boolean hasAppended = false;
+		while (toAppend.length() > 512) {
+			if (hasAppended)
+				result.append(" ");
+			hasAppended = true;
+			result.append(toAppend.substring(0, 510)).append(newline);
+			toAppend = toAppend.substring(510);
+		}
+		if (hasAppended)
+			result.append(" ");
+		result.append(toAppend).append(newline);
+		return toAppend;
+	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((entries == null) ? 0 : entries.hashCode());
```

Comment:
Why don't we consider the other field values in hashcode or equals?

Reply:
The old code hasn't done as well. Not sure if equals/hashCode is really needed for our code. I just wanted to change the semantic as minimal as possible. Of course two manifests with the same content but one dirty the other not are not considered equal this way ... what do you think? I would stay with current code but if you think otherwise I can easaly change. Removing the two methods or adding the remaining fields would be the options.